support/utils/command.go

```diff
@@ -34,41 +35,57 @@ var commandRunner = gexe.New()
 // be set in the in the invoker to make sure the right path is used for the binaries while invoking
 // rest of the workfow after this helper is triggered.
 func FindOrInstallGoBasedProvider(pPath, provider, module, version string) (string, error) {
-	if commandRunner.Prog().Avail(pPath) != "" {
+	if gexe.ProgAvail(pPath) != "" {
 		log.V(4).InfoS("Found Provider tooling already installed on the machine", "command", pPath)
 		return pPath, nil
 	}
 
+	var stdout, stderr bytes.Buffer
 	installCommand := fmt.Sprintf("go install %s@%s", module, version)
 	log.V(4).InfoS("Installing provider tooling using go install", "command", installCommand)
-	p := commandRunner.RunProc(installCommand)
-	if p.Err() != nil {
-		return "", fmt.Errorf("failed to install %s: %s", pPath, p.Err())
+	p := commandRunner.NewProc(installCommand)
+	p.SetStdout(&stdout)
+	p.SetStderr(&stderr)
+	result := p.Run()
+	if result.Err() != nil {
+		return "", fmt.Errorf("failed to install %s: %s: \n %s", pPath, result.Result(), stderr.String())
 	}
 
-	if !p.IsSuccess() || p.ExitCode() != 0 {
-		return "", fmt.Errorf("failed to install %s: %s", pPath, p.Result())
+	if !result.IsSuccess() || result.ExitCode() != 0 {
+		return "", fmt.Errorf("failed to install %s: %s \n %s", pPath, result.Result(), stderr.String())
 	}
 
-	if providerPath := commandRunner.Prog().Avail(provider); providerPath != "" {
+	log.V(4).InfoS("Installed provider tooling using go install", "command", installCommand, "output", stdout.String())
+
+	if providerPath := gexe.ProgAvail(provider); providerPath != "" {
 		log.V(4).Infof("Installed %s at %s", pPath, providerPath)
 		return provider, nil
 	}
 
-	p = commandRunner.RunProc("ls $GOPATH/bin")
-	if p.Err() != nil {
-		return "", fmt.Errorf("failed to install %s: %s", pPath, p.Err())
+	p = commandRunner.NewProc("ls $GOPATH/bin")
```

Comment:
Do you think `gexe` could be improved to make output caputure easier ? If so, please open an issue.